From c3c96e7032077fdfd48023d05f1b544de49d88de Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Mon, 8 Mar 2021 10:56:40 +0700
Subject: [PATCH 145/358] pinctrl: sh-pfc: r8a77990: Add missing TPU pinctrl

This patch adds support missing TPU pins, pin functions and pin groups
for the R8A77990 and R8A774C0 SoC.

Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 drivers/pinctrl/sh-pfc/pfc-r8a77990.c | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/drivers/pinctrl/sh-pfc/pfc-r8a77990.c b/drivers/pinctrl/sh-pfc/pfc-r8a77990.c
index b70eb04..d6ca947 100644
--- a/drivers/pinctrl/sh-pfc/pfc-r8a77990.c
+++ b/drivers/pinctrl/sh-pfc/pfc-r8a77990.c
@@ -2824,6 +2824,21 @@ static const unsigned int tpu0_to1_mux[] = {
 	TPU0TO1_MARK,
 };
 
+static const unsigned int tpu0_to2_pins[] = {
+	/* TPU0TO2 */
+	RCAR_GP_PIN(6, 15),
+};
+static const unsigned int tpu0_to2_mux[] = {
+	TPU0TO2_MARK,
+};
+static const unsigned int tpu0_to3_pins[] = {
+	/* TPU0TO3 */
+	RCAR_GP_PIN(6, 16),
+};
+static const unsigned int tpu0_to3_mux[] = {
+	TPU0TO3_MARK,
+};
+
 /* - QSPI0 ------------------------------------------------------------------ */
 static const unsigned int qspi0_ctrl_pins[] = {
 	/* QSPI0_SPCLK, QSPI0_SSL */
@@ -3827,7 +3842,7 @@ static const unsigned int vin5_clk_b_mux[] = {
 };
 
 static const struct {
-	struct sh_pfc_pin_group common[255];
+	struct sh_pfc_pin_group common[257];
 #ifdef CONFIG_PINCTRL_PFC_R8A77990
 	struct sh_pfc_pin_group automotive[21];
 #endif
@@ -3977,6 +3992,8 @@ static const struct {
 		SH_PFC_PIN_GROUP(pwm6_b),
 		SH_PFC_PIN_GROUP(tpu0_to0),
 		SH_PFC_PIN_GROUP(tpu0_to1),
+		SH_PFC_PIN_GROUP(tpu0_to2),
+		SH_PFC_PIN_GROUP(tpu0_to3),
 		SH_PFC_PIN_GROUP(qspi0_ctrl),
 		SH_PFC_PIN_GROUP(qspi0_data2),
 		SH_PFC_PIN_GROUP(qspi0_data4),
@@ -4389,6 +4406,8 @@ static const char * const pwm6_groups[] = {
 static const char * const tpu0_groups[] = {
 	"tpu0_to0",
 	"tpu0_to1",
+	"tpu0_to2",
+	"tpu0_to3",
 };
 
 static const char * const qspi0_groups[] = {
-- 
2.7.4

